data/adapters/gwas_adapter.py

```diff
@@ -0,0 +1,263 @@
+
+        super(GWAS, self).__init__()
+
+    # trying to capture the breakline problem described in the comments above
+    def line_appears_broken(self, row):
+        return row[-1].startswith('"[') and not row[-1].endswith(']"')
+
+    def edge_keys(self, row):
+        variant_id = 'variants/' + \
+            build_variant_id(row[4], row[5], row[6], row[7])
+
+        ontology_term_id = 'ontology_terms/'
+        equivalent_term_id = None
+        # give preference to MONDO if defined, otherwise, use EFO term
+        if row[1] != 'NA':
+            ontology_term_id = ontology_term_id + row[1]
+            equivalent_term_id = 'ontology_terms/' + row[2]
+        else:
+            ontology_term_id = ontology_term_id + row[2]
+
+        # MANY records have no ontology term. Ignoring those lines.
+        if ontology_term_id == 'ontology_terms/':
+            return (None, None, None, None)
+
+        key = hashlib.sha256(
+            (variant_id + '_' + ontology_term_id).encode()).hexdigest()
+
+        return (key, variant_id, ontology_term_id, equivalent_term_id)
+
+    def process_file(self):
+        header = None
+        trying_to_complete_line = None
+
+        print('Collecting genes...')
+        genes = self.get_genes_from_variant_to_genes_file()
+
+        print('Collecting tagged variants...')
+        tagged = self.get_tagged_variants()
+
+        # Many records are duplicated with different tagged values.
+        # We are collecting all tagged values at once.
+        # For that, we need to keep track of which keys we already processed to avoid duplicated entries.
+        processed_keys = set()
+
+        parsed_data_file = open(self.output_filepath, 'w')
+
+        print('Processing file...')
+
+        for record in open(self.to_ontology_filepath, 'r'):
+            if header is None:
+                header = record.strip().split('\t')
+                continue
+
+            if trying_to_complete_line:
+                record = trying_to_complete_line + record
+                trying_to_complete_line = None
+
+            row = record.strip().split('\t')
+
+            if self.line_appears_broken(row):
+                trying_to_complete_line = record
+                continue
+
+            key, variant_id, ontology_term_id, equivalent_term_id = self.edge_keys(
+                row)
+
+            if key is None or key in processed_keys:
+                continue
+
+            # a few rows are incomplete. Filling empty values with None
+            row = row + [None] * (len(header) - len(row))
+
+            props = {
+                '_key': key,
+                '_from': variant_id,
+                '_to': ontology_term_id,
+                'genes': genes.get(row[0]),
+                'equivalent_ontology_term': equivalent_term_id,
+                'study_id': row[3],
+                'lead_chrom': row[4],
+                'lead_pos': row[5],
+                'lead_ref': row[6],
+                'lead_alt': row[7],
```

Comment:
There is an interesting denormalization here; these values should always match the _from field.  Not sure what to do with this information, but it's true.